src/GitHub.TeamFoundation.14/Services/VSGitServices.cs

```diff
@@ -81,25 +83,68 @@ public string GetLocalClonePathFromGitProvider()
             bool recurseSubmodules,
             object progress = null)
         {
-#if TEAMEXPLORER14
-            var gitExt = serviceProvider.GetService<IGitRepositoriesExt>();
-            gitExt.Clone(cloneUrl, clonePath, recurseSubmodules ? CloneOptions.RecurseSubmodule : CloneOptions.None);
+            var teamExplorer = serviceProvider.TryGetService<ITeamExplorer>();
+            Assumes.Present(teamExplorer);
 
-            // The operation will have completed when CanClone goes false and then true again.
-            await gitExt.WhenAnyValue(x => x.CanClone).Where(x => !x).Take(1);
-            await gitExt.WhenAnyValue(x => x.CanClone).Where(x => x).Take(1);
+#if TEAMEXPLORER14
+            await StartClonenOnConnectPageAsync(teamExplorer, cloneUrl, clonePath, recurseSubmodules);
+            NavigateToHomePage(teamExplorer); // Show progress on Team Explorer - Home
+            await WaitForCloneOnHomePageAsync(teamExplorer);
+            vsServices.Value.TryOpenRepository(clonePath); // Show the repository on Team Explorer - Home
 #else
             var gitExt = serviceProvider.GetService<IGitActionsExt>();
             var typedProgress = ((Progress<ServiceProgressData>)progress) ?? new Progress<ServiceProgressData>();
+            typedProgress.ProgressChanged += (s, e) => statusBar.Value.ShowMessage(e.ProgressText);
+            var cloneTask = gitExt.CloneAsync(cloneUrl, clonePath, recurseSubmodules, default(CancellationToken), typedProgress);
 
-            await Microsoft.VisualStudio.Shell.ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                typedProgress.ProgressChanged += (s, e) => statusBar.Value.ShowMessage(e.ProgressText);
-                await gitExt.CloneAsync(cloneUrl, clonePath, recurseSubmodules, default(CancellationToken), typedProgress);
-            });
+            NavigateToHomePage(teamExplorer); // Show progress on Team Explorer - Home
+            await cloneTask;
 #endif
         }
 
+        static async Task StartClonenOnConnectPageAsync(
```

Comment:
Oh, that's actually by design. Is so that I can test the alternative code path in Visual Studio 2017, before double checking that it works the same in 2015.

I think it will make testing easier in future if we light up new features in 2017, but don't exclude that is designed to run inside 2015. We could have an option to run inside 2017, but with the alternative code path used inside 2015.